airbyte-integrations/connectors/source-cart/source_cart/source.py

```diff
@@ -61,6 +61,17 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             return False, repr(e)
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+
+        # try to check an end_date value. It can be used for different CI tests
+        end_date = config.get("end_date")
+        if end_date:
```

Comment:
we should do this check in 'check method'

Reply:
And also in the streams method, just because the check method does not share his output to the streams

Reply:
I've added a decorator for this validation